### ISSUE TITLE
Mars Flyby accomplished?

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#324]
I did a Mars Flyby two or three turns ago, then just now flew my first mission with the XMS-2.  Among the prestige firsts & seconds it displayed "MARS FLYBY".  Should it do that?

![Mars Flyby](https://user-images.githubusercontent.com/3118014/74970193-cf0a0780-53d2-11ea-89e6-52136a55ff79.png)

**Comment by rnyoakum** [comment]
That shouldn't be showing up. Nor should the two-person & three-person craft prestige firsts.

**Comment by peyre** [comment]
Huh.  I always assumed that flying Apollo also gives you credit for the next level down (2-person craft), and XMS-2 would give you credit for 3-person craft and 2-person craft - that is, it's cumulative.  In this case those are prestige seconds, since the Soviets had already flown Voskhod and Soyuz.  So that part might be by design (or might not, I'm not sure. I just assumed).  The Mars Flyby was unexpected though!

**Comment by rnyoakum** [comment]
Okay, I assumed that only the two-person craft would give you the two-person craft award, etc. Looking at prest.cpp, however, it seems that the mini-shuttle includes the three-person craft, the three-person craft goal awards the two-person craft goal, and the two-person craft goal awards the one-person craft goal. I'd thought otherwise, but it makes its own kind of sense.

With that out of the way, the only problem appears to be the Mars Flyby. I'm guessing the prestige goals set from completing the satellite mission aren't getting cleared properly because there isn't a mission summary screen when the Flyby completes. I'll look into that.

While researching the chained craft prestige awards, I noticed something curious. In the Set_Goal function in prest.cpp, ln. 571-2 (commit 63d49f0aa9c65828a56123aac89d78b98919aac3) read:
```
    case Prestige_FourPerson:
        return (sum + Set_Goal(plr, Prestige_Minishuttle, 1));
```
This suggests the four-person craft also awards the mini-shuttle prestige award - and then the three-person craft, etc. - which **doesn't** make logical sense, as the four-person craft is definitely not a mini-shuttle. It follows via the "give a reward for all preceding craft" approach, but I still don't like it. I haven't haven't confirmed this in-game, since I've never invested in the Jupiter / LK-700 Capsule, but it's worth checking.

**Comment by peyre** [comment]
Of course!  The Mars Flyby shows up in the next mission summary window because it doesn't appear in a mission summary of its own.  (Same would go for Jupiter & Saturn as well of course.)  That meshes with something else: I've looked once or twice at the Director Prestige graph after a successful M/J/S Flyby and it doesn't seem to give you prestige then - which would make sense if the game only applies prestige in the mission summary screen.  If so, that's worth fixing, since it's kind of unfair to further delay an already-delayed payoff on those missions.  Not hugely important, but would be nice to fix.

I agree, the direct ascent shouldn't award the minishuttle; that doesn't make sense.  I wonder, though, if it was intentional: that DA was supposed to give you a lot of prestige, and they did that by making it cumulative with the shuttle.  Regardless, I think it might be good to take that out; it doesn't really make sense.

**Comment by peyre** [comment]
Hey, here's another one!  I skipped the lunar probe landing and did the Manned Lunar Pass, and that gave me credit for the probe landing, even though I've never even prototyped the landing probe.  Of course, this one could be sidestepped by making the probe landing a milestone for the manned landing rather than the lunar pass.

![Lunar Pass   Lunar Probe Landing](https://user-images.githubusercontent.com/3118014/75617190-a7244d80-5b10-11ea-9dd6-623805bffb80.png)

**Comment by peyre** [comment]
(FWIW, if there were some implementation of #175 that would sidestep this issue, at least for the outer planet probes.  Only thing is, I'm not sure where would be the best place to squeeze in a view of the mission results screen.)

**Comment by peyre** [comment]
Hey, here's an idea - how about if at the end of the turn, after any missions flow and before the next turn's newscast appears, a Mission Review screen like this pops up for the Mars/Jupiter/Saturn mission?  That way you'd see (and get) the prestige gain/loss and you wouldn't miss news of your flyby because you flipped quickly through the newscast 'cause you were in a hurry to get on with the turn?  You could also choose to Replay the mission if you want to see the movie of the flyby.  That would take care of #175 and eliminate these showing in an unrelated mission's review screen.
